ext/ruby_http_parser/ruby_http_parser.c

```diff
@@ -316,15 +316,34 @@ VALUE Parser_initialize(int argc, VALUE *argv, VALUE self) {
   ParserWrapper *wrapper = NULL;
   DATA_GET(self, ParserWrapper, wrapper);
 
-  wrapper->header_value_type = rb_iv_get(CLASS_OF(self), "@default_header_value_type");
+  VALUE default_keyword_value_type = Qnil;
+
+  if (RB_TYPE_P(argv[argc-1], T_HASH)) {
+    static ID keyword_ids[1];
+
+    if (!keyword_ids[0]) {
+      keyword_ids[0] = rb_intern("default_header_value_type");
```

Comment:
It should be safe because it's the same value and re-assignment must not cause any side-effect at any timing. But yes, I agree that it's not needed to be `static` in this case.
I'll update the change.